package.json

```diff
@@ -13,25 +13,22 @@
     "Lorenzo Stoakes <lorenzo@resin.io>"
   ],
   "scripts": {
-    "lint": "resin-lint angular.coffee core.coffee request.coffee test",
-    "pretest": "npm run lint && npm run-script prepublish",
+    "lint": "resin-lint test",
+    "pretest": "npm run lint && npm run-script prepublishOnly",
     "test": "mocha",
-    "prepublish": "coffee -c angular.coffee core.coffee request.coffee"
+    "prepublish": "require-npm4-to-publish",
+    "prepublishOnly": "tsc"
   },
   "license": "MIT",
-  "dependencies": {
-    "bluebird": "^3.0.6",
-    "bluebird-lru-cache": "^1.0.0",
-    "lodash": "^4.0.0",
-    "request": "^2.82.0",
-    "typed-error": "^0.1.0"
-  },
+  "dependencies": {},
   "devDependencies": {
-    "@types/bluebird": "^3.5.12",
-    "@types/request": "^2.0.3",
+    "bluebird": "^3.0.6",
     "chai": "^3.3.0",
-    "coffee-script": "~1.11.0",
+    "coffee-script": "~1.12.7",
```

Comment:
Maybe it's worth updating coffeescript to 2.0.0 ?

Reply:
2.0.0 is a breaking changing that I'd really rather not deal with - it can also be part of a later patch PR if it offers any benefit since it's only used for tests (because tests have to break the type system in order to test some error cases)